pkg/cvo/availableupdates_test.go

```diff
@@ -0,0 +1,317 @@
+package cvo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-version-operator/pkg/clusterconditions"
+)
+
+type queueStub struct{}
+
+func (q queueStub) Add(item interface{}) {}
+
+func (q queueStub) Len() int {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (q queueStub) Get() (item interface{}, shutdown bool) {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (q queueStub) Done(item interface{}) {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (q queueStub) ShutDown() {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (q queueStub) ShutDownWithDrain() {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (q queueStub) ShuttingDown() bool {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (q queueStub) AddAfter(item interface{}, duration time.Duration) {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (q queueStub) AddRateLimited(item interface{}) {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (q queueStub) Forget(item interface{}) {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (q queueStub) NumRequeues(item interface{}) int {
+	// TODO implement me
+	panic("implement me")
+}
+
+type notFoundProxyLister struct{}
+
+func (n notFoundProxyLister) List(selector labels.Selector) (ret []*configv1.Proxy, err error) {
+	return []*configv1.Proxy{}, nil
+}
```

Comment:
Tell that to my code generator! :P

Reply:
Yeah... My IDE generator is unaware of the actual kube client contract (whether the `List` is supposed to return empty slice or a `nil` when there are no resources) and any proper consumer will need to handle both so...

Also note that the `List` here is not even properly anywhere, it just needs to exist so that the fake implements the `ProxyLister` interface. If we needed a proper kube client fake then we would use https://pkg.go.dev/k8s.io/client-go/testing .

Reply:
I ended up cleaning the generated code even more: removed todos and unused param / return value names, and return nils instead of empty slices.